src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java

```diff
@@ -24,11 +21,8 @@
     @PostMapping(value = "/transactions/{transactionId}/company-accounts",
             produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount companyAccount,
-            HttpServletRequest request)
-            throws NoSuchAlgorithmException {
-        Transaction transaction = (Transaction) request.getSession().getAttribute(AttributeName.TRANSACTION.getValue());
-        CompanyAccount result = companyAccountService
-                .save(companyAccount, transaction.getCompanyNumber());
+            HttpServletRequest request) {
+        CompanyAccount result = companyAccountService.createCompanyAccount(companyAccount);
         return ResponseEntity.status(HttpStatus.CREATED).body(result);
```

Comment:
we cannot hard code a successful response code, we have to validate the creation of the company accounts resource